src/support/bits.h

```diff
@@ -66,14 +66,14 @@ int CountLeadingZeroes(T v) {
   return CountLeadingZeroes(typename std::make_unsigned<T>::type(v));
 }
 
-template <typename T>
-inline static T RotateLeft(T val, T count) {
+template <typename T, typename U>
```

Comment:
Sorry I missed this before. I'm not sure having a second type is good here because they could end up being different sizes if we're not careful, and then the bit-twiddling doesn't work. If the warning is about negation of an unsigned value, maybe we could just cast it to signed for that expression instead.

If we did that, I think we wouldn't need #474 either.